plugins/okta/komand_okta/util/api.py

```diff
@@ -151,22 +150,37 @@ def make_request(self, method: str, url: str, json_data: dict = None, params: di
             )
 
             if response.status_code == 400:
-                raise PluginException(preset=PluginException.Preset.BAD_REQUEST, data=response.text)
+                raise PluginException(
+                    preset=PluginException.Preset.BAD_REQUEST,
+                    data=response_data.format(status_code=response.status_code, text=response.text),
```

Comment:
are we returning status_code as a new field now in the exception?

Reply:
This feels a little messy passing the status code back through a PluginException in the string and then using regex to extract it. Can you remove PluginExceptions from the API layer and return a new error type, for example BadRequestError (or AuthenticationError for auth-related issues), and pass information back through that to the consuming code? The consuming code can then catch that BadRequestError, pull the status code property off it, then send a response back with the proper state.

Reply:
I added a custom exception from which we can get the status code returned by the API.